src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1181,13 +1184,17 @@ CheckCitusVersion(int elevel)
  * this function logs an error with the specified elevel and returns false,
  * otherwise it returns true.
  */
-static bool
+bool
 CheckAvailableVersion(int elevel)
 {
 	char *availableVersion = NULL;
 
-	Assert(CitusHasBeenLoaded());
-	Assert(EnableVersionChecks);
+	if (citusVersionKnownCompatible ||
```

Comment:
This check is for calling CheckAvailableVersion directly in ErrorIfUnstableCreateOrAlterExtensionStmt. We changed ErrorIfUnstableCreateOrAlterExtensionStmt to call this function instead of CheckCitusVersions, because CheckCitusVersions also perform checks for installed version, which we should not perform during ALTER EXTENSION (Because during  ALTER EXTENSION, it is expected that version will mismatch).